microprofile-metrics/README.adoc

```diff
@@ -0,0 +1,1422 @@
+
+[source,options="nowrap"]
+----
+curl -H "Accept: application/json" http://localhost:9990/metrics/application/checksTimer
+----
+
+[source,json]
+----
+{
+    "checksTimer": {
+        "p99": 0.136153,
+        "min": 0.01075,
+        "max": 0.136153,
+        "mean": 0.07816790480018097,
+        "p50": 0.088464,
+        "p999": 0.136153,
+        "stddev": 0.05164982729810127,
+        "p95": 0.136153,
+        "p98": 0.136153,
+        "p75": 0.136153,
+        "fiveMinRate": 0.4595570030187892,
+        "fifteenMinRate": 0.5489683372380184,
+        "meanRate": 0.03506919414327886,
+        "count": 3,
+        "oneMinRate": 0.1581582828694362
+    }
+}
+----
+
+== Gauge
+
+Gauge is the most variable metric which is fully up to the application. It is
+required to define the metric type for this kind of metric. Let's update
+the `PrimeNumberChecker`:
+
+[source,java]
+----
+package org.quickstart.microprofile.metrics;
+
+import org.eclipse.microprofile.metrics.MetricUnits;
+import org.eclipse.microprofile.metrics.annotation.Counted;
+import org.eclipse.microprofile.metrics.annotation.Gauge;
+import org.eclipse.microprofile.metrics.annotation.Timed;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Path("/")
+@ApplicationScoped
+public class PrimeNumberChecker {
+
+    private long highestPrimeNumberSoFar = 2;
+
+    @GET
+    @Path("/prime/{number}")
+    @Produces(MediaType.TEXT_PLAIN)
+    @Counted(name = "performedChecks", displayName="Performed Checks", description = "How many prime checks have been performed.")
+    @Timed(name = "checksTimer", absolute = true, description = "A measure of how long it takes to perform the primality test.", unit = MetricUnits.MILLISECONDS)
+    public String checkIfPrime(@PathParam("number") long number) {
+        if (number < 1) {
+            return "Only natural numbers can be prime numbers.";
+        }
+
+        if (number == 1) {
+            return "1 is not prime.";
+        }
+
+        if (number == 2) {
+            return "2 is prime.";
+        }
+
+        if (number % 2 == 0) {
+            return number + " is not prime, it is divisible by 2.";
+        }
+
+        for (int i = 3; i < Math.floor(Math.sqrt(number)) + 1; i = i + 2) {
+            if (number % i == 0) {
+                return number + " is not prime, is divisible by " + i + ".";
+            }
+        }
+
+        if (number > highestPrimeNumberSoFar) {
+            highestPrimeNumberSoFar = number;
+        }
+
+        return number + " is prime.";
+    }
+
+    @Gauge(name = "highestPrimeNumberSoFar", unit = MetricUnits.NONE, description = "Highest prime number so far.")
+    public Long highestPrimeNumberSoFar() {
+        return highestPrimeNumberSoFar;
+    }
+}
+----
+
+WARNING: The metric unit for gauges is required.
```

Comment:
is it? you just specified `@Gauge(name = "highestPrimeNumberSoFar", unit = MetricUnits.NONE`,...)

Reply:
I meant there is no default. I will update the description.